tests/dist/mpi/test_multi_tenant.cpp

```diff
@@ -0,0 +1,49 @@
+#include <catch2/catch.hpp>
+
+#include "fixtures.h"
+
+#include <faabric/scheduler/Scheduler.h>
+
+namespace tests {
+
+/* 12/11/2021 - This test deliberately fails as multi-tenant MPI execution is
+ * still not supported. We keep it here to showcase this is the use case we
+ * want to support.
```

Comment:
Rather than commenting out tests like this, we should use the catch2 `[.]` tag which means they aren't run by default. That way they will still get compiled.